docs/pages/setup/guides/terraform-provider.mdx

```diff
@@ -0,0 +1,236 @@
+---
+title: Terraform Provider
+description: Configuration as a code with Terraform Provider
+---
+
+This section will cover:
+
+- Configuring OpenSSH server `sshd` to join a Teleport cluster. Existing fleets of
+  OpenSSH servers can be configured to accept SSH certificates dynamically issued by a Teleport CA.
+- Configuring OpenSSH client `ssh` to login into nodes inside a Teleport
+  cluster.
+
+## Prerequisites
+
+- [Terraform >= (=terraform.version=)+](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+- Teleport (=teleport.version=) Cloud, Open Source or Enterprise
+
+```bash
+terraform version
+Terraform v(=terraform.version=)
+```
+
+```bash
+tctl version
+Teleport v(=teleport.version=) go(teleport.golang)
+```
+
+Create a folder `teleport-terraform` to hold some temporary files:
+
+```bash
+mkdir -p teleport-terraform
+```
+
+## Step 1/4. Install terraform provider
+
+<Tabs>
+  <TabItem label="Linux">
+  ```bash
+  mkdir -p ${HOME?}/.terraform.d/plugins/gravitational.com/teleport/teleport/(=teleport.version=)/linux_amd64
+  curl -L -O https://get.gravitational.com/terraform-provider-teleport-v(=teleport.version=)-linux-amd64-bin.tar.gz
+  tar -zxvf terraform-provider-teleport-v(=teleport.version=)-linux-amd64-bin.tar.gz -C ${HOME?}/.terraform.d/plugins/gravitational.com/teleport/teleport/(=teleport.version=)/linux_amd64
+  ```
+  </TabItem>
+
+  <TabItem label="MacOS">
+  ```bash
+  mkdir -p ${HOME?}/.terraform.d/plugins/gravitational.com/teleport/teleport/(=teleport.version=)/darwin_amd64
+  ```
+  </TabItem>
+</Tabs>
+  
+# Step 2/4. Create a terraform user
+
+Put the following content into `terraform.yaml`:
+
+```
+kind: role
+metadata:
+  name: terraform
+spec:
+  allow:
+    rules:
+      - resources: ['user', 'role', 'token', 'trusted_cluster', 'github', 'oidc', 'saml']
+        verbs: ['list','create','read','update','delete']
+version: v3
+---
+kind: user
+metadata:
+  name: terraform
+spec:
+  roles: ['terraform']
+version: v2
+```
+
+Run:
+
+```bash
+tctl create terraform.yaml
+tctl auth sign --format=file --user=terraform --out=terraform-identity --ttl=10h
+```
+
+<Admonition type="note">
```

Comment:
This is helpful - I like having a note to address a specific pitfall one might encounter.

One thought: I wonder if a "drop-down note" might be helpful for these specific kinds of items. Click to get the more expansive instructions otherwise you can skip over it and go on to the next step?

Reply:
Good point, can you create this as a feature request to gravitational/next repo, I will discuss with @alexwolfe later

Reply:
Added: https://github.com/gravitational/next/issues/298